src/typecode/contenttype.py

```diff
@@ -285,8 +289,13 @@ def is_archive(self):
         """
         Return True if the file is some kind of archive or compressed file.
         """
-        # FIXME: we should use extracode archive detection
+
         # TODO: also treat file systems as archives
+
+        y = self.can_extract(location)
+        if can_extract(location) == True:
+            return can_extract(location)
+
```

Comment:
What is `y` here? Why not using it? and why such a meaningless name for a variable that you do not use? And what is `self.can_extract` ? it does not exists afaik.... Did you test your code? You also need to add new unit tests of course.
Something like this is likely all you need:

```
extractible = can_extract(location)
if extractible:
    return extractible
```

Reply:
Or may be rather:
Add this cached attribute above at https://github.com/nexB/scancode-toolkit/pull/869/files#diff-c41ee25999eeb86203e67767ef5115c7R183
```
self._is_archive = None
```

Then this all you need is IMHO no more than this for the whole method:
```
    @property
    def is_archive(self):
        """
        Return True if the file is some kind of archive or compressed file.
        """
        # TODO: also treat file systems as archives
        if self._is_archive is None:
            self._is_archive = can_extract(location)
        return self._is_archive
```

And of course we need a lot of (possibly new) unit tests like this one: https://github.com/nexB/scancode-toolkit/blob/a9083191a04f62c05d588a22fa8f4839eeffc79d/tests/typecode/test_contenttype.py#L156

Reply:
where do I write the test cases? 
In the same file contenttype.py? 
I am not familiar with writing tests. Thanks.